konan_cli/main.py

```diff
@@ -176,31 +176,104 @@ def build(image_name, config_file, dry_run, verbose):
 
     # build image
     image, build_logs = local_config.build_image(image_tag=image_name)
-
+    local_config.latest_built_image = image.tags[0]
+    local_config.save()
     # TODO: use low-level api to stream logs realtime
     if verbose:
         for chunk in build_logs:
             if 'stream' in chunk:
                 for line in chunk['stream'].splitlines():
                     click.echo(line)
 
-# @konan.command()
-# @click.pass_context
-# def test():
-#     """
-#     tbd
-#     """
-#     pass
-
 
 # @konan.command()
 # @click.pass_context
-# def publish():
+# def test():
 #     """
 #     tbd
 #     """
 #     pass
 
+# TODO: use sdk to fetch KCR creds
+@click.option('--image-tag', help="name of the generated image", required=False)
+@konan.command()
+def publish(image_tag):
+    """
+    Publish image built to konan container registry
+    """
+    if not global_config.access_token:
+        login()
+
+    # Getting KCR creds if not found
+    if not (global_config.token_name and global_config.token_password):
+        response = requests.get(url=f"{global_config.API_URL}/registry/token/",
+                                headers={'content-type': 'application/json',
+                                         'Authorization': f'Bearer {global_config.access_token}'})
+        if response.ok:
+            r_json = response.json()
+            global_config.token_name = r_json['token_name']
+            global_config.token_password = r_json['token_password']
+            global_config.save()
+        else:
+            # Refresh if access token is expired
+            if response.status_code == 401:
+                refresh_response = requests.get(url=f"{global_config.API_URL}/api/auth/token/refresh/",
+                                                headers={'content-type': 'application/json'})
+                if refresh_response.ok:
+                    global_config.access_token = refresh_response.json()['access']
+                    global_config.save()
+                    # Resend KCR creds request in case of access token is expired
+                    response = requests.get(url=f"{global_config.API_URL}/registry/token/",
+                                            headers={'content-type': 'application/json',
+                                                     'Authorization': f'Bearer {global_config.access_token}'})
+                    if response.ok:
+                        r_json = response.json()
+                        global_config.token_name = r_json['token_name']
+                        global_config.token_password = r_json['token_password']
+                        global_config.save()
+                else:
+                    click.echo(
+                        "Error fetching token_name and token_password for konan container registry, please try to re-login")
```

Comment:
Could we change this to 
```
Looks like you're not logged in. Run `konan login` first then try again.
```
The user doesn't need to know any details of token or password or container registry.